lang/cs/Org.Apache.REEF.Common/metrics/Api/AbstractMetric.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Common.metrics.Api
+{
+    /// <summary>
+    /// Abstract immutable metric class. All metrics put in the record by the MetricsSource are 
+    /// kept as <see cref="IEnumerable{T}" />.
+    /// </summary>
+    public abstract class AbstractMetric : IMetricsInfo
```

Comment:
what's the reasoning for the abstract class? Why not make it an interface with a default implementation?